### PR TITLE
Owner close pool

### PR DESCRIPTION
**Pull request by sujine2**
Owner can close the pool

If the pool is closed,

- Unable to restart
- Staking is not possible
- Withdrawal, getting reward is possible
- Reward is not increase anymore.

### CODE REVIEW

File: contracts/interface/IStakingPoolV2.sol [hunk 1]
```diff
@@ -2,21 +2,22 @@
...
-  error NotInRound();
   error StakingNotInitiated();
   error InvalidAmount();
   error ZeroReward();
   error OnlyAdmin();
   error NotEnoughPrincipal(uint256 principal);
   error ZeroPrincipal();
-  error IsOpened();
+  error IsFinished();
+  error IsClosed();
```
Comment (qpzm) [inline contracts/interface/IStakingPoolV2.sol]: How about `Finished()` and `Closed()`?  I usually use the prefix, `is`, with boolean values.

File: test/claim.test.ts [hunk 2]
```diff
@@ -119,4 +131,81 @@ describe('StakingPool.claim', () => {
...
+
+      await advanceTime(10);
+
+      const poolDataAfter = await getPoolData(testEnv);
+      const userDataAfter = await getUserData(testEnv, alice);
+
+      expect(poolDataAfter).to.eql(poolDataBefore);
+      expect(userDataAfter).to.eql(userDataBefore);
+    });
+  });
```
Comment (qpzm) [inline test/claim.test.ts]: I added a test case here!

File: contracts/StakingPoolV2.sol [hunk 3]
```diff
@@ -4,6 +4,7 @@ import './logic/StakingPoolLogicV2.sol';
 import './interface/IStakingPoolV2.sol';
 import './token/StakedElyfiToken.sol';
 import '@openzeppelin/contracts/token/ERC20/IERC20.sol';
+import 'hardhat/console.sol';
```
Comment (qpzm) [inline contracts/StakingPoolV2.sol]: Please delete unused imports.

File: contracts/logic/StakingPoolLogicV2.sol [hunk 4]
```diff
@@ -3,6 +3,7 @@ pragma solidity 0.8.4;
 import '../StakingPoolV2.sol';
 import '@openzeppelin/contracts/token/ERC20/IERC20.sol';
 import '@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol';
+import 'hardhat/console.sol';
```
Comment (qpzm) [inline contracts/logic/StakingPoolLogicV2.sol]: Please delete unused imports

File: contracts/interface/IStakingPoolV2.sol (hunk 1 above)
Comment (donguks) [reply contracts/interface/IStakingPoolV2.sol]: Here's a link for the naming conventions: https://docs.soliditylang.org/en/v0.8.11/style-guide.html#naming-conventions.

File: contracts/logic/StakingPoolLogicV2.sol [hunk 5]
```diff
@@ -38,9 +39,11 @@ library StakingPoolLogicV2 {
     view
     returns (uint256)
   {
+
```
Comment (donguks) [inline contracts/logic/StakingPoolLogicV2.sol]: Please delete this empty line.

File: test/claim.test.ts [hunk 6]
```diff
@@ -119,4 +131,81 @@ describe('StakingPool.claim', () => {
...
+          firstRoundStartTimestamp,
+          firstRoundInit.duration
+        );
+      await testEnv.stakingAsset.connect(alice).faucet();
+      const tx = await testEnv.stakingAsset.connect(alice).approve(testEnv.stakingPool.address, RAY);
+      await advanceTimeTo(await getTimestamp(tx), firstRoundStartTimestamp);
+      await testEnv.stakingPool.connect(alice).stake(amount);
+    });
+
+    it('pool is closed and alice claim reward', async () => {
```
Comment (donguks) [inline test/claim.test.ts]: I think `pool is closed` is redundant. How about wrapping the statement like below?
```
context('the pool is closed', async () => {})

or

beforeEach('the pool is closed', async () => {}) ~~
```

**Review by qpzm — APPROVED** [review]

**Pull request merged**